torch/testing/_internal/common_utils.py

```diff
@@ -816,6 +816,25 @@ def settings(*args, **kwargs):
 except ImportError:
     print('Fail to import hypothesis in common_utils, tests are not derandomized')
 
+
+slow_tests_dict: Optional[Dict[str, float]] = None
+def check_slow_test_from_stats(test):
+    global slow_tests_dict
+    if slow_tests_dict is None:
+        url = 'https://raw.githubusercontent.com/janeyx99/test-infra/main/stats/.pytorch-slow-tests'
+        contents = urlopen(url, timeout=1).read().decode('utf-8')
```

Comment:
How often does this get run? From the looks of it, it seems like this would get re-downloaded for every single test file in a given test run; is that correct?

Reply:
Yes, but so would the disabled test file in check_disable below. I'm not sure how much we would save by getting both files separately, but I could do that in a separate PR.

Reply:
adding to the conversation. `timeout=1` seems like a pretty big overhead if there's something wrong. worth taking a look in follow up PRs